### PR TITLE
Assessments controller tests

### PR DESCRIPTION
**Pull request by caleybrock**
Tests for the two functions in `assessments_controller`.

`section_responses` still isn't as clean as it could be. This API already existed for the most part and has been improved from the original version (some functions have been moved to a model and tested). While I think there is still more to do here, at least now this new version  is tested.
These tests were modeled off of: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/api_controller_test.rb#L206

The `index` function, which is new and did not exist in previous versions of the assessments API, is also now tested. This function is pretty clean, with most of the work done in tested functions on models.

I had already written authentication tests for this controller, these are mostly about expected output.

### CODE REVIEW

File: dashboard/test/controllers/api/v1/assessments_controller_test.rb [hunk 1]
```diff
@@ -39,22 +50,188 @@ class Api::V1::AssessmentsControllerTest < ActionController::TestCase
...
+
+    # Set up an assessment for that script.
+    sub_level1 = create :text_match, name: 'level_free_response', type: 'TextMatch'
+    sub_level2 = create :multi, name: 'level_multi_unsubmitted', type: 'Multi'
+    sub_level3 = create :multi, name: 'level_multi_correct', type: 'Multi'
+    sub_level4 = create :multi, name: 'level_multi_incorrect', type: 'Multi'
+    sub_level5 = create :multi, name: 'level_multi_unattempted', type: 'Multi'
+
+    level1 = create :level_group, name: 'LevelGroupLevel1', type: 'LevelGroup'
+    level1.properties['title'] =  'Long assessment 1'
```
Comment (maddiedierker) [inline dashboard/test/controllers/api/v1/assessments_controller_test.rb]: nit: extra space after `=`

File: dashboard/test/controllers/api/v1/assessments_controller_test.rb [hunk 2]
```diff
@@ -39,22 +50,188 @@ class Api::V1::AssessmentsControllerTest < ActionController::TestCase
...
+    level1.properties['pages'] = [{levels: ['level_free_response', 'level_multi_unsubmitted']}, {levels: ['level_multi_correct', 'level_multi_incorrect']}, {levels: ['level_multi_unattempted']}]
```
Comment (maddiedierker) [inline dashboard/test/controllers/api/v1/assessments_controller_test.rb]: nit: break this into multiple lines (same for `expected_answers` and `expected_questions` below)

File: dashboard/test/controllers/api/v1/assessments_controller_test.rb [hunk 3]
```diff
@@ -39,22 +50,188 @@ class Api::V1::AssessmentsControllerTest < ActionController::TestCase
...
+
+    # Call the controller method.
+    get :section_responses, params: {
+      section_id: @section.id,
+      script_id: script.id
+    }
+
+    assert_response :success
+
+    # All these are translation missing because we don't actually generate i18n files in tests.
```
Comment (maddiedierker) [inline dashboard/test/controllers/api/v1/assessments_controller_test.rb]: you should be able to write to a fake i18n locale to test these strings:

https://github.com/code-dot-org/code-dot-org/blob/7729b35c919ccc91119147293ae3d7e1229f54c3/dashboard/test/models/script_test.rb#L1074-L1090

File: dashboard/test/controllers/api/v1/assessments_controller_test.rb [hunk 4]
```diff
@@ -39,22 +50,188 @@ class Api::V1::AssessmentsControllerTest < ActionController::TestCase
...
+    sub_level1 = create :text_match, name: 'level_free_response', type: 'TextMatch'
+    sub_level2 = create :multi, name: 'level_multi_unsubmitted', type: 'Multi'
+    sub_level3 = create :multi, name: 'level_multi_correct', type: 'Multi'
+    sub_level4 = create :multi, name: 'level_multi_incorrect', type: 'Multi'
+    create :multi, name: 'level_multi_unattempted', type: 'Multi'
+
+    level1 = create :level_group, name: 'LevelGroupLevel1', type: 'LevelGroup'
+    level1.properties['title'] =  'Long assessment 1'
+    level1.properties['anonymous'] = 'true'
+    level1.properties['pages'] = [{levels: ['level_free_response', 'level_multi_unsubmitted']}, {levels: ['level_multi_correct', 'level_multi_incorrect']}, {levels: ['level_multi_unattempted']}]
```
Comment (maddiedierker) [inline dashboard/test/controllers/api/v1/assessments_controller_test.rb]: nit: break this onto multiple lines

i think you could also set all of the `level1` properties in a single assignment like:
```
level1.properties = {
  title: 'Long assessment 1',
  anonymous: true,
  ...
}
```

Comment (caleybrock) [reply dashboard/test/controllers/api/v1/assessments_controller_test.rb]: Chatted offline and decided this wasn't necessary right now.

**Pull request merged**